Classes/DownloadHandler.cs

```diff
@@ -190,7 +190,7 @@ private void DownloadProgThread()
 
                 try
                 {
-                    saveLocation = FileUtils.GetSaveFolder();
+                    saveLocation = FileUtils.GetSaveFolder(this.progInfo);
```

Comment:
I think it would be better if we just passed around the progid rather than the whole programme class.